tracing/zipkin/zipkin.go

```diff
@@ -259,6 +312,12 @@ func FromContext(ctx context.Context) (*Span, bool) {
 	return span, true
 }
 
+// CollectorToContext stores a Zipkin collector to be used when annotating
+// resources using their own child spans with NewChildSpan.
+func CollectorToContext(ctx context.Context, c Collector) context.Context {
+	return context.WithValue(ctx, CollectorContextKey, c)
+}
```

Comment:
FWIW, that's not an idiomatic use of `net.Context`. From the [docs](https://godoc.org/golang.org/x/net/context):

> Use context Values only for request-scoped data that transits processes and APIs, not for passing optional parameters to functions.

In [Go OpenTracing implementation](https://github.com/opentracing/opentracing-go/blob/master/span.go#L90) the Span exposes a reference to the Tracer

```go
	// Provides access to the Tracer that created this Span.
	Tracer() Tracer
```